packages/components/src/components/domainproperties/dataset/DatasetDesignerPanels.tsx

```diff
@@ -232,44 +289,94 @@ export class DatasetDesignerPanelImpl extends React.PureComponent<Props & Inject
         const { setSubmitting } = this.props;
         const { model, fileImportData } = this.state;
 
-        saveDomain(model.domain, model.getDomainKind(), model.getOptions(), model.name)
-            .then(response => {
-                this.setState(
-                    produce((draft: Draft<State>) => {
-                        const updatedModel = draft.model;
-                        updatedModel.exception = undefined;
-                        updatedModel.domain = response;
-                    }),
-                    () => {
-                        // If we're importing Dataset file data, import file contents
-                        if (fileImportData) {
-                            this.handleFileImport();
-                        } else {
-                            const { model } = this.state;
+        let keyPropIndex;
+        let visitPropIndex;
+
+        if (this._participantId && this._sequenceNum) {
+            // filter out these fields
+            const updatedDomain = model.domain.merge({
+                fields: model.domain.fields
+                    .filter(field => field.name != this._participantId)
+                    .filter(field => field.name != this._sequenceNum)
+                    .toList(),
+            }) as DomainDesign;
+
+            updatedDomain.fields.map((field, index) => {
+                if (model.keyPropertyName && field.name === model.keyPropertyName) {
+                    keyPropIndex = index;
+                }
+                if (model.visitDatePropertyName && field.name === model.visitDatePropertyName) {
+                    visitPropIndex = index;
+                }
+            });
+
+            const updatedModel = produce(model, (draft: Draft<DatasetModel>) => {
+                draft.domain = updatedDomain;
+            });
+
+            this.setState(
+                produce((draft: Draft<State>) => {
+                    draft.keyPropertyIndex = keyPropIndex;
+                    draft.visitDatePropertyIndex = visitPropIndex;
+                }),
+                () =>
+                    saveDomain(
+                        updatedModel.domain,
```

Comment:
here is where I would use `updatedDomain` instead of `updatedModel` and just use model.getDomainKind(), model.getOptions(), model.name as the other params.

Reply:
let me know if the next commit to this one isn't correct.